src/models/flinkArtifact.ts

```diff
@@ -61,6 +62,19 @@ export class FlinkArtifactTreeItem extends TreeItem {
     this.iconPath = new ThemeIcon(resource.iconName);
     this.description = resource.description;
 
-    // TODO: add tooltip
+    this.tooltip = createFlinkArtifactToolTip(resource);
   }
 }
+
+export function createFlinkArtifactToolTip(resource: FlinkArtifact): vscode.MarkdownString {
+  return new vscode.MarkdownString(
+    [
+      `**${resource.name}**`,
+      resource.description,
+      `Cloud: ${resource.provider}`,
+      `Region: ${resource.region}`,
+    ]
+      .filter(Boolean)
+      .join("\n\n"),
+  );
+}
```

Comment:
Let's use [`CustomMarkdownString.resourceTooltip()`](https://github.com/confluentinc/vscode/blob/341677b465226314dc1acefec9666b99c55ef839/src/models/main.ts#L84-L92) for consistent function and styling. Example:
https://github.com/confluentinc/vscode/blob/341677b465226314dc1acefec9666b99c55ef839/src/models/flinkStatement.ts#L282-L295